File: Program/LawCreator/bin/test/lawcreator/xml/TypeTest.js
Clazz.declarePackage ("test.lawcreator.xml");
Clazz.load (null, "test.lawcreator.xml.TypeTest", ["pl.wroc.pwr.student.lawcreator.xml.TypeImpl"], function () {
c$ = Clazz.decorateAsClass (function () {
this.type = null;
Clazz.instantialize (this, arguments);
}, test.lawcreator.xml, "TypeTest");
Clazz.defineMethod (c$, "setUp", 
function () {
this.type =  new pl.wroc.pwr.student.lawcreator.xml.TypeImpl ();
});
Clazz.defineMethod (c$, "gettingAndSettingFieldsTest", 
function () {
var name = "name";
var pattern = "pattern";
this.type.setName (name);
this.type.setPattern (pattern);
org.junit.Assert.assertEquals (name, this.type.getName ());
org.junit.Assert.assertEquals (pattern, this.type.getPattern ());
});
Clazz.defineMethod (c$, "addingEnumerationsTest", 
function () {
var enum1 = "enum1";
var enum2 = "enum2";
this.type.add (enum1);
this.type.add (enum2);
org.junit.Assert.assertEquals (2, this.type.getEnumerations ().size ());
org.junit.Assert.assertEquals (enum1, this.type.getEnumerations ().get (0));
org.junit.Assert.assertEquals (enum2, this.type.getEnumerations ().get (1));
});
});

File: Program/LawCreator/test/test/lawcreator/xml/TypeTest.java
package test.lawcreator.xml;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import pl.wroc.pwr.student.lawcreator.xml.Type;
import pl.wroc.pwr.student.lawcreator.xml.TypeImpl;

public class TypeTest {
	Type type;

	@Before
	public void setUp() throws Exception {
		type = new TypeImpl();
	}

	@Test
	public void gettingAndSettingFieldsTest() {
		String name = "name";
		String pattern = "pattern";
		
		type.setName(name);
		type.setPattern(pattern);
		
		assertEquals(name, type.getName());
		assertEquals(pattern, type.getPattern());
	}
	
	@Test
	public void addingEnumerationsTest() {
		String enum1 = "enum1";
		String enum2 = "enum2";
		
		type.add(enum1);
		type.add(enum2);
		
		assertEquals(2, type.getEnumerations().size());
		assertEquals(enum1, type.getEnumerations().get(0));
		assertEquals(enum2, type.getEnumerations().get(1));
	}

}
